charts/nidx/templates/shards-api-deploy.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nidx-shards-api
```

Comment:
I would rename the charts to `nidx-api`. Even if we have multiple API's we can deploy all of them in a single pod/process.